templates/e3sm_diags.bash

```diff
@@ -51,6 +51,15 @@ cd climo
 cp -s ${climoDir}/${case}_*_${Y1}??_${Y2}??_climo.nc .
 cd ..
 
+{%- if "diurnal_cycle" in sets %}
+# Create local links to input diurnal cycle climo files
+climoDir={{ output }}/post/atm/{{ grid }}/clim/{{ '%dyr' % (year2-year1+1) }}
```

Comment:
I wasn't sure how to actually change the `clim` directory to something mentioning the diurnal cycle.  Perhaps using the `prefix` variable?

The current code does manage to create separate `tmp` sub-directories for regular `climo` and diurnal cycle `climo`